chain/network/src/routing/routing_table_actor.rs

```diff
@@ -511,23 +516,21 @@ impl Handler<RoutingTableMessages> for RoutingTableActor {
                 )
             }
             RoutingTableMessages::RoutingTableUpdate { prune, prune_edges_not_reachable_for } => {
-                let edges_removed = if self.needs_routing_table_recalculation {
-                    self.recalculate_routing_table_and_maybe_prune_edges(
-                        prune,
-                        prune_edges_not_reachable_for,
-                    )
-                } else {
-                    Vec::new()
-                };
-                self.needs_routing_table_recalculation = false;
+                let mut edges_removed =
+                    if self.needs_routing_table_recalculation || prune == Prune::PruneNow {
```

Comment:
so this is the place where the bug was, right?

and the refactoring/splitting of the methods was just to improve the readability?
(if yes - please mention it in PR description - as I was looking for the bug in those methods instead)

Reply:
Thanks. Yes, that was the bug. I updated the PR.